### DISCUSSION TITLE
Generate static css files for each theme

### DISCUSSION THREAD

**Discussion by hmd-ai** [saadeghi/daisyui#3048] [Ideas / feature requests]
Firstly, thank you for making daisyui. I specially love the wide selection of themes.

This might be somewhat out of scope, but I was wondering if it's possible to generate static css files for each theme (and/or maybe even each component). This would make it possible to use daisyui without having to have any special build setup, i.e. just drop the theme css file and use classes similar to other component based css libraries. I realise this is not aligned with tailwind's philosophy, but daisyui is a component based library (and uses tailwind as utility based library internally).

I imagine this could be done during daisyui's build step to generate the static files.

Here's how I think it should work for the end user:

```tsx

import React from 'react';
import 'daisyui/themes/cupcake.css'
import 'daisyui/btn.css'

export const StyledButton = () => {
  return (
    <button className="btn btn-primary">
      daisyUI Button
    </button>
  )
}

```

Or just pure html and css:

```html
<head>
<link rel="stylesheet" type="text/css" href="daisyui/cupcake/all.css">
</head>

....
<a class="btn btn-primary">Button</a>
```

**Comment by saadeghi** [comment]
This is one of the things I'm planning to do for daisyUI 5. But first we should wait for Tailwind CSS 4 to be released and see the API changes and how is it going to use plugins in the CSS file instead of `tailwind.config.js` file. Then, based on that API, it will be possible to provide raw CSS files for each component and **maybe** there will be no need for a build step.

**Reply by hmd-ai** [reply]
I had a look at tailwind css v4.0.0-alpha, as you said many of the configs are moved to css file. 
do you plan to release an alpha version of daisyui 5 based on v4.0.0-alpha? (I'm happy to test the early version)
